src/ripple/app/tx/impl/XChainBridge.cpp

```diff
@@ -0,0 +1,1627 @@
+
+    if (attBegin->createCount <= claimCount)
+    {
+        return tecXCHAIN_ACCOUNT_CREATE_PAST;
+    }
+    if (attBegin->createCount >= claimCount + 128)
+    {
+        // Limit the number of claims on the account
+        return tecXCHAIN_ACCOUNT_CREATE_TOO_MANY;
+    }
+
+    STXChainBridge::ChainType const dstChain =
+        STXChainBridge::otherChain(srcChain);
+
+    STXChainBridge::ChainType const attDstChain =
+        STXChainBridge::dstChain(attBegin->wasLockingChainSend);
+
+    if (attDstChain != dstChain)
+    {
+        return tecXCHAIN_WRONG_CHAIN;
+    }
+
+    auto const claimKeylet =
+        keylet::xChainCreateAccountClaimID(bridgeSpec, attBegin->createCount);
+
+    // sleCID may be null. If it's null it isn't created until the end of this
+    // function (if needed)
+    auto const sleCID = psb.peek(claimKeylet);
+    bool createCID = false;
+    if (!sleCID)
+    {
+        createCID = true;
+
+        // Check reserve
+        auto const balance = (*sleDoor)[sfBalance];
+        auto const reserve =
+            psb.fees().accountReserve((*sleDoor)[sfOwnerCount] + 1);
+
+        if (balance < reserve)
+            return tecINSUFFICIENT_RESERVE;
+    }
+
+    std::vector<AttestationBatch::AttestationCreateAccount> atts;
+    atts.reserve(std::distance(attBegin, attEnd));
+    for (auto att = attBegin; att != attEnd; ++att)
+    {
+        if (!signersList.count(calcAccountID(att->publicKey)))
+            continue;
+        atts.push_back(*att);
+    }
+    if (atts.empty())
+    {
+        return tecXCHAIN_PROOF_UNKNOWN_KEY;
+    }
+
+    XChainCreateAccountAttestations curAtts = [&] {
+        if (sleCID)
+            return XChainCreateAccountAttestations{
+                sleCID->getFieldArray(sfXChainCreateAccountAttestations)};
+        return XChainCreateAccountAttestations{};
+    }();
+
+    auto const rewardAccounts = curAtts.onNewAttestations(
+        &atts[0], &atts[0] + atts.size(), quorum, signersList);
+
+    if (!createCID)
+    {
+        // Modify the object before it's potentially deleted, so the meta data
+        // will include the new attestations
+        if (!sleCID)
+            return tecINTERNAL;
+        sleCID->setFieldArray(
+            sfXChainCreateAccountAttestations, curAtts.toSTArray());
+        psb.update(sleCID);
+    }
+
+    // Account create transactions must happen in order
+    if (rewardAccounts && claimCount + 1 == attBegin->createCount)
+    {
+        auto const r = finalizeClaimHelper(
+            psb,
+            bridgeSpec,
+            attBegin->toCreate,
+            /*dstTag*/ std::nullopt,
+            doorAccount,
+            attBegin->sendingAmount,
+            /*rewardPoolSrc*/ doorAccount,
+            attBegin->rewardAmount,
+            *rewardAccounts,
+            srcChain,
+            sleCID,
+            OnTransferFail::removeClaim,
+            ctx_.journal);
+        if (!isTesSuccess(r))
```

Comment:
I think `finalizeClaimHelper`  will return an error if the `sleCID` was never created (when we get all needed attestations in one batch) , because the  `psb.remove()` call will fail.

Reply:
The code looks correct as-is. `sleCID` will be null in this case, and the `finalizeClaimHelper` checks for null before the `remove` function is called.